django/http/multipartparser.py

```diff
@@ -145,6 +149,11 @@ def parse(self):
         old_field_name = None
         counters = [0] * len(handlers)
 
+        # To count the number of keys in the request.
+        num_post_keys = 0
+        # To limit the amount of data read from the request.
+        read_size = None
```

Comment:
It'd be sufficient to set this to `settings.DATA_UPLOAD_MAX_MEMORY_SIZE + 1` here...